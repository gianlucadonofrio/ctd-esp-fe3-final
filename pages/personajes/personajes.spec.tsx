import { render } from "@testing-library/react";
import CharacterPage from "./[id].page";
import { character } from "dh-marvel/test/mocks/character";

describe("Personajes page", () => {
  const setComics = jest.fn();

  it("should render successfully", () => {
    const { container } = render(<CharacterPage character={character} />);
    expect(container).toMatchSnapshot();
  });

  it("should set comics", () => {
    setComics(character.comics.items);

    expect(setComics).toHaveBeenCalled();
  });
});
