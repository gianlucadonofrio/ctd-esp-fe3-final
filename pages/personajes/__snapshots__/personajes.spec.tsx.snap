// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personajes page should render successfully 1`] = `
<div>
  <section
    class="css-fmr9wc-MuiStack-root"
  >
    <img
      alt="Abomination (Ultimate)"
      class="MuiBox-root css-1gfgts2"
      src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
    />
    <h1
      class="MuiTypography-root MuiTypography-h4 css-g9veo7-MuiTypography-root"
    >
      Abomination (Ultimate)
    </h1>
    <div
      class="MuiBox-root css-1lb6n3v"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </section>
</div>
`;
